### ISSUE TITLE
Circular dependency detected

### ISSUE THREAD

**Issue by nixegend** [unshiftio/url-parse#49]
Hi, I have found these circular dependencies in console using plugin "circular-dependency-plugin".

```
ERROR in Circular dependency detected:
node_modules\url-parse\index.js -> node_modules\url-parse\lolcation.js -> node_modules\url-parse\index.js

ERROR in Circular dependency detected:
node_modules\url-parse\lolcation.js -> node_modules\url-parse\index.js -> node_modules\url-parse\lolcation.js
```

**Comment by 3rd-Eden** [comment]
The circular dependency is lazy loaded so this will never be an issue.

**Comment by grantila** [comment]
"never be an issue" is obviously not really true if the circular dependency requires everybody with this plugin to `exclude: /node_modules\/url-parse/`. It may affect more static analyzers too.

In a relatively large project I'm working on now (where `npm ls | wc -l` is 1156), this and `readable-stream` *are the only packages* causing this. Surely the code could be written for static analyzers to not be fooled. They can't _know_ it's gonna be fine at run-time.

The fact that the code won't fail run-time doesn't necessarily mean that it's not an issue, if developers (incl. those not directly using this package!) need workarounds for their builds to go through. @lpinca's idea sounds good.

**Comment by 3rd-Eden** [comment]
Published the fix in 1.1.9.
